dydxprotocol/v4-web
# feat: Implement solana deposit flow

## How to review
Ignore the first commit as that is addressed by: https://github.com/dydxprotocol/v4-web/pull/923/files

## Recording


https://github.com/user-attachments/assets/82d51fe8-0d65-433a-a6cb-c5e999738fb0



## How to test
Solana deposit options are only implemented on mainnet and when using skip. You must configure a `solanaRpcUrl` in env.json and make sure the `useSkip` feature flag is true. In addition, you must enable skip svm flags on `v4-abacus` in `configs/V4StateManagerConfigs.kt`

## Changes
- public/configs/cctp.json
    - add solana USDC token addresses to this config file
- public/configs/v1/env.json
    - add an entry for the solana rpc url for fetching balances from the chain
- src/hooks/useAccountBalance.ts
    - updated to support account balances
- src/hooks/useEndpointsConfig.ts
    - updated to retrieve solana rpc url from config
- src/hooks/usePhantomWallet.ts
    - added a method for signing a transaction
- src/hooks/useSolanaBalance.ts
    - responsible for getting balances from the solana block chain
- src/hooks/useSolanaConnection.ts
   - responsible for instantiating the solana rpc connection
- src/views/forms/AccountManagementForms/DepositForm.tsx
   - Updated so that when Phantom is connected the default chain is solana
   - Uses phantom to sign solana transactions
- src/views/forms/AccountManagementForms/DepositForm/DepositButtonAndReceipt.tsx
   - ignore wagmi checks when using phantom
- src/views/forms/AccountManagementForms/SourceSelectMenu.tsx
   - filters for solana chains when using Phantom
